services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -210,6 +211,9 @@ public Response updateUser(final UserRepresentation rep) {
         } catch (ReadOnlyException re) {
             session.getTransactionManager().setRollbackOnly();
             throw ErrorResponse.error("User is read only!", Status.BAD_REQUEST);
+        } catch (PasswordPolicyNotMetException e) {
+            session.getTransactionManager().setRollbackOnly();
+            throw ErrorResponse.error("Password policy not met", new Object[] { e.getMessage() }, Status.BAD_REQUEST);
```

Comment:
Instead of adding the details to the ErrorResponse, it is preferred to just log them. For instance something like:
```
logger.warnf("Password policy not met for user %s. Details: %s"m user.getUsername(), e.getMessage());
```
We don't want to add details to error response for security purposes.

Reply:
Thanks for replying. We will add logging.

It's the admin api and you need a valid user to create or update a user, otherwise you get an error "HTTP 401 Unauthorized". So I see no risk in answering with a more detailed answer. The reason for rejecting the password is also shown in the webinterface.

In our case, we can give the user a concrete indication of what is wrong.

Reply:
BTV. If those endpoints are called on behalf of the user himself, it is better to use Account REST API (or account console) instead of admin REST API, which is supposed to be for administrators.

Reply:
But if the passsword is in hands of an attacker, you have other problems than XSS/CSRF?

I'm sorry, but I don't understand why you show a very concrete message to the user via the web interface (you also need a username and password), but not via the API? I do not see the difference.

![image](https://github.com/keycloak/keycloak/assets/116587968/d088618e-bcd7-46cf-a4c5-37e6a9686c53)